.github/workflows/build-&-publish-docker-image.yml

```diff
@@ -1,8 +1,13 @@
 name: build-&-publish-docker-image
 
+concurrency:
+  group: "${{ github.ref }}"
+  cancel-in-progress: true
+
 on:
   push:
     branches: [ staging ]
+  pull_request:
```

Comment:
any reason why this shouldn't publish on PR like blobber @cnlangzi ?

Reply:
@anish-squareops  says it make too heavy tasks on our github runner. I would suggest to enable it too.